src/simplewallet/simplewallet.cpp

```diff
@@ -4830,6 +4845,25 @@ bool simple_wallet::register_service_node(const std::vector<std::string> &args_)
     return true;
   }
 
+  try
+  {
+    const auto& response = m_wallet->get_service_node_list();
+    bool found = false;
+    for (const auto& snode_info : response.service_node_states)
+      if (string_to_public_key(snode_info.service_node_pubkey) == service_node_key)
+        found = true;
```

Comment:
Early break out.

But not necessary. I've updated the RPC call to accept a service node key for it to look for so I'll update this after the updated RPC call is added in.